runtime/win32.c

```diff
@@ -419,19 +420,29 @@ CAMLexport int caml_read_directory(wchar_t * dirname,
   wchar_t * template;
   intptr_t h;
   struct _wfinddata_t fileinfo;
+  int res;
 
   dirnamelen = wcslen(dirname);
   if (dirnamelen > 0 &&
       (dirname[dirnamelen - 1] == L'/'
        || dirname[dirnamelen - 1] == L'\\'
        || dirname[dirnamelen - 1] == L':'))
     template = caml_stat_wcsconcat(2, dirname, L"*.*");
-  else
+  else {
     template = caml_stat_wcsconcat(2, dirname, L"\\*.*");
+    dirnamelen++;
```

Comment:
We only increment `dirnamelen` on the `else` branch: If `dirname` already ends with `/`, `\` or `:`, there is no need to change it. Do you think it would be better to introduce a new variable `xdirnamelen` and set it on both branches?

Reply:
Oh dear, I'd misread this - what you had before (the `dirname++`) is better than the `xdirnamelen`, yes. Possibly worth adding a comment `dirnamelen++; /* template[dirname] always points after the backlash */`